libvmi/read.c

```diff
@@ -97,13 +97,13 @@ vmi_read(
 
         if(dtb) {
             paddr = vmi_pagetable_lookup(vmi, dtb, start_addr + buf_offset);
+            if (!paddr) {
```

Comment:
I considered changing it to use `vmi_pagetable_lookup_extended` directly, but that also uses `!info->paddr` to determine it's return status right now, so that issue affects far more code paths.